src/basic/uid-classification.c

```diff
@@ -109,23 +110,52 @@ const UGIDAllocationRange *acquire_ugid_allocation_range(void) {
         return &default_ugid_allocation_range;
 }
 
-bool uid_is_system(uid_t uid) {
-        const UGIDAllocationRange *defs;
-        assert_se(defs = acquire_ugid_allocation_range());
+bool uid_in_range(uid_t uid, UGIDRangeFlags flags) {
+        assert(flags);
 
-        return uid <= defs->system_uid_max;
+        if (FLAGS_SET(flags, UGID_RANGE_DYNAMIC))
+                if (DYNAMIC_UID_MIN <= uid && uid <= DYNAMIC_UID_MAX)
+                        return true;
+
+        if (FLAGS_SET(flags, UGID_RANGE_CONTAINER))
+                if (CONTAINER_UID_BASE_MIN <= uid && uid <= CONTAINER_UID_BASE_MAX)
+                        return true;
+
+        if (FLAGS_SET(flags, UGID_RANGE_SYSTEM)) {
+                const UGIDAllocationRange *defs;
+                assert_se(defs = acquire_ugid_allocation_range());
+
+                if (uid <= defs->system_uid_max)
+                        return true;
+        }
+
+        return false;
+}
+
+bool gid_in_range(gid_t gid, UGIDRangeFlags flags) {
+
+        if (FLAGS_SET(flags, UGID_RANGE_SYSTEM)) {
+                const UGIDAllocationRange *defs;
+                assert_se(defs = acquire_ugid_allocation_range());
+
+                if (gid <= defs->system_gid_max)
+                        return true;
+        }
+
+        return uid_in_range((uid_t) gid, UPDATE_FLAG(flags, UGID_RANGE_SYSTEM, 0));
```

Comment:
uh, UPDATE_FLAG we use to update a variable, and do so depending on some dynamic value but here you jsut want to mask off a static value, hence just do `&=~`...

I think much cleaner would be if two functions would call a third one for the common parts, instead of this masking business.

but again, i am not convinced we want this at all.